src/studio/src/designer/frontend/shared/features/dataModelling/DataModelling.tsx

```diff
@@ -65,8 +65,8 @@ function DataModelling(props: IDataModellingContainerProps): JSX.Element {
 
   const onSchemaSelected = setSelectedModelMetadata;
 
-  const onSaveSchema = (schema: any) => {
-    const $id = sharedUrls().dataModelsApi + selectedModelMetadata?.value?.repositoryRelativeUrl;
+  const onSaveSchema = (schema: any, name: string) => {
+    const $id = sharedUrls().dataModelsApi + (selectedModelMetadata?.value?.repositoryRelativeUrl || `/App/models/${name}.schema.json`);
```

Comment:
I'm not sure if the name is set correctly in this instance, but selectedMetadata.label should be synced with the name you are saving, so when the fetchDatamodelsMetadata is called it fetches the name that matches the label.
Maybe the name in the Schema-editor should never be a local state, and just read from props?